### ISSUE TITLE
Ability to disable the component

### ISSUE THREAD

**Issue by zwilson2** [BrightspaceUI/file-uploader#49]
In the case where it takes time to process the file, the user may want to disable the file upload control until the processing is complete.

**Comment by dbatiste** [comment]
I think there should be better integration between the UI (file input) and the upload.  I don't think the consuming code of this component should be left to manage the state (for example, while uploading, in the event of errors, etc.).

**Comment by zwilson2** [comment]
I'm not just talking while this component is processing the file though. Our FRA takes the uploaded file and pushes it to S3 which can be a fairly lengthy process depending on how large the file is. This component wouldn't be aware that the S3 upload is happening so our component that wraps this would need to be able to control the disabled state.

**Comment by dbatiste** [comment]
Yep.  I think the application code should not be acting as the controller between the file upload and the file input.  There's a whole bunch of scenarios that can come up related to upload (to S3 or where ever) that potentially affect the state of the UI, and we don't want the app code to know about all of that.
